scripts/e2e-test.py

```diff
@@ -28,7 +28,8 @@ def main() -> None:
     request_amount = 123
 
     token = l2_contracts["MintableToken"]
-    token.functions.mint(deployer.address, request_amount).transact()
+    tx_hash = token.functions.mint(deployer.address, request_amount).transact()
+    web3.eth.wait_for_transaction_receipt(tx_hash)
```

Comment:
Do we really need this? There's the approve afterwards, and as the nonces increase this should wait implicitly.

Reply:
> Do we really need this? There's the approve afterwards, and as the nonces increase this should wait implicitly.

I thought it is just good practice. Theoretically, it is possible that the mint tx is not included in a block, and the approve is then reverted, which is wasteful, right?